src/osdc/Objecter.cc

```diff
@@ -2732,7 +2740,13 @@ int Objecter::_calc_target(op_target_t *t, epoch_t *last_force_resend,
     } else {
       int osd;
       bool read = is_read && !is_write;
-      if (read && (t->flags & CEPH_OSD_FLAG_BALANCE_READS)) {
+      if (t->use_osd_epoch) {
+	assert(read);
+	// XXX: Should we check that osd is part of acting?
```

Comment:
i don't think so. take the replicated pool as an example, it is collecting the scrub maps from the actingbackfill set which in turn comes from the "up" set. so we are fixing the "living" objects instead of the "acting" objects.